src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ProcedureRequestMapperTest.java

```diff
@@ -1,20 +1,121 @@
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertAll;
 
+@SuppressWarnings("checkstyle:magicnumber")
+@ExtendWith(MockitoExtension.class)
 class ProcedureRequestMapperTest {
 
+    @InjectMocks
+    private ProcedureRequestMapper procedureRequestMapper;
+    @Mock
+    private UUIDGenerator uuidGenerator;
+
+    @Test
+    void testMapMessageToProcedureRequestNoPatientClinicalInfo() {
+        final Message message = new Message(Collections.emptyList());
+
+        final var procedureRequest = procedureRequestMapper.mapToProcedureRequest(message);
+
+        assertThat(procedureRequest).isEmpty();
+    }
+
+    @Test
+    void testMapToProcedureRequestThrowsExceptionWhenFreeTextIsMissing() {
+        final Message message = new Message(List.of(
+            "S10+10",
+            "CIN+UN"
+        ));
+
+        assertThatThrownBy(() -> procedureRequestMapper.mapToProcedureRequest(message))
+            .isInstanceOf(FhirValidationException.class)
+            .hasMessage("Unable to map message. "
+                + "The FreeText segment is mandatory in Clinical Information");
+    }
+
+    @Test
+    void testMapToProcedureRequestThrowsExceptionWhenTypeOfClinicalObservationIsMissing() {
+        final Message message = new Message(List.of(
+            "S10+10",
+            "FTX+CID+++COELIAC"
+        ));
+
+        assertThatThrownBy(() -> procedureRequestMapper.mapToProcedureRequest(message))
+            .isInstanceOf(MissingSegmentException.class)
+            .hasMessage("EDIFACT section is missing segment CIN");
+    }
+
+    @Test
+    void testMapToProcedureRequestFreeText() {
+        final Message message = new Message(List.of(
+            "S10+10",
+            "CIN+UN",
+            "FTX+CID+++COELIAC"
+        ));
+
+        final var procedureRequest = procedureRequestMapper.mapToProcedureRequest(message).get();
+
+        assertThat(procedureRequest.getNote()).hasSize(1)
+            .satisfies(note -> assertThat(note.get(0).getText()).isEqualTo("COELIAC"));
+    }
+
     @Test
-    void testMapMessageToProcedureRequest() {
-        final Message message = new Message(new ArrayList<>());
+    void testMapToProcedureRequestFreeTexts() {
+        final Message message = new Message(List.of(
+            "S10+10",
+            "CIN+UN",
+            "FTX+CID+++COELIAC",
+            "FTX+CID+++ON AZATHIOPRINE",
+            "FTX+CID+++GASTRIC ULCER DECLINE"
+        ));
+
+        var procedureRequest = procedureRequestMapper.mapToProcedureRequest(message).get();
+
+        assertThat(procedureRequest.getNote()).hasSize(3)
+            .satisfies(note -> assertAll(
+                () -> assertThat(note.get(0).getText()).isEqualTo("COELIAC"),
+                () -> assertThat(note.get(1).getText()).isEqualTo("ON AZATHIOPRINE"),
+                () -> assertThat(note.get(2).getText()).isEqualTo("GASTRIC ULCER DECLINE")
+            ));
```

Comment:
In this case we can use `.containsExactly()` if you want
```suggestion
        assertThat(procedureRequest.getNote()).hasSize(3)
            .extracting(Annotation::getText)
            .containsExactly("COELIAC", "ON AZATHIOPRINE", "GASTRIC ULCER DECLINE");
```
This works because order is preserved by `extracting` (it might be `map` if extracting doesn't exist - can't remember)

Reply:
This is a very good suggestion! Fixed in 1a93e6c5b9a1c73b438f8bddf548dd64434deb88
Thanks!